### DISCUSSION TITLE
How to discourage hotlinking or direct download of assets?

### DISCUSSION THREAD

**Discussion by kara-todd** [vercel/vercel#5738] [Help]
I am trying to self host a web font which requires "adequate technical protection measures" to ensure that the fonts can only be used on the fonts they are licensed for.

Based on some poking around I feel like this approach (using .htaccess) is reasonable:

```
#http://www.fontshop.com/blog/newsletters/pdf/webfontfontuserguide.pdf

RewriteEngine On
RewriteCond %{HTTP:Origin}  !^$|http(s)?://(www\.)?example\.com$ [NC]
RewriteRule \.(woff|eot)$ - [NC,L]
RewriteCond %{HTTP_REFERER} !.
RewriteRule \.(woff|eot)$ - [F,NC,L]
Options -Indexes
```

How would I create something like this on Vercel? Essentially, I think I just want to check the request is coming from a specific set of whitelisted domains and forbid access if not.

**Comment by jacobhq** [comment]
You could use cloudflare hotlink protection.
